libpod/container.go

```diff
@@ -177,6 +172,13 @@ type containerState struct {
 	// ExtensionStageHooks holds hooks which will be executed by libpod
 	// and not delegated to the OCI runtime.
 	ExtensionStageHooks map[string][]spec.Hook `json:"extensionStageHooks,omitempty"`
+
+	// Special container state attributes for Linux
+	containerStateLinux
+	// Special container state attributes for Windows
+	containerStateWindows
+	// Special container state attributes for Darwin
+	containerStateDarwin
```

Comment:
This is the approach taken [by the runtime-spec][1] to allow for validating a Windows config from a Linux box, etc.  I doubt we need that sort of flexibility, in which case I think it might make more sense to have a single `platformState` or some such here, with each build target defining their own `platformState` structure with whatever they need.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.1/specs-go/config.go#L22-L27

Reply:
@wking My concern would be DB errors as we try to unmarshall, say, a Windows spec on Linux. Seemingly unlikely, but it seems like a good way to blow up the DB in spectacular fashion.

Reply:
> My concern would be DB errors as we try to unmarshall, say, a Windows spec on Linux. Seemingly unlikely, but it seems like a good way to blow up the DB in spectacular fashion.

Sound good right?  Dying with "I do not match this database" is certainly better than "ah, that unmarshall's fine.  And look, the Linux state is empty.  Hmm.  [crash later]".  If you want to guard against folks putting a Windows DB on a Linux box, then stick a flag in the DB somewhere with the target OS and check that before loading?

Reply:
We already have a bunch of init-level checks for other DB misconfigurations (you can't use a DB with a different container store, for example, you wouldn't be able to see the old set of containers) so this seems like a good way of dealing with the issue.

Reply:
> We already have a bunch of init-level checks for other DB misconfigurations...

I've filed #1033 with this pivot, but I'm not quite sure how to phrase the DB check.  Can you give me some hints in that PR?